lib/Dialect/IR/TcpTilingInterfaceImpl.cpp

```diff
@@ -0,0 +1,164 @@
+    return getOpFoldResultAsValue(ofr, b, loc);
+  });
+  return vals;
+}
+
+struct SliceOpTiling
+    : public TilingInterface::ExternalModel<SliceOpTiling, tcp::SliceOp> {
+
+  SmallVector<utils::IteratorType> getLoopIteratorTypes(Operation *op) const {
+    auto sliceOp = cast<tcp::SliceOp>(op);
+    return SmallVector(sliceOp.getType().getRank(),
+                       utils::IteratorType::parallel);
+  }
+
+  SmallVector<Range> getIterationDomain(Operation *op, OpBuilder &b) const {
+    // The iteration domain describes the loop bounds for the tiled operation.
+    // For a slice op, we keep the loop bounds same as the output of the slice.
+    // We can compute the indices of the input tensor from the iteration indices
+    // of the output.
+    //
+    // So, the iteration domain is always [0, size, 1] for every dimension.
+    auto sliceOp = cast<tcp::SliceOp>(op);
+    OpFoldResult zero = b.getIndexAttr(0);
+    OpFoldResult one = b.getIndexAttr(1);
+    SmallVector<Range> loopRanges(sliceOp.getType().getRank(),
+                                  {zero, one, one});
+    for (auto [idx, size] : llvm::enumerate(sliceOp.getSizes())) {
+      loopRanges[idx].offset = zero;
+      loopRanges[idx].size = size;
+    }
+    return loopRanges;
+  }
+
+  FailureOr<TilingResult>
+  getTiledImplementation(Operation *op, OpBuilder &b,
+                         ArrayRef<OpFoldResult> offsets,
+                         ArrayRef<OpFoldResult> sizes) const {
+    // getTiledImplementation and generateResultTileValue are the same
+    // because:
+    // 1. `getIterationDomain` returns a loop with zero offsets
+    //     and the result slice length
+    // 2. `getResultTilePosition` returns the same offsets and sizes as
+    //     the input offsets and slices.
+    return generateResultTileValue(op, b, 0, offsets, sizes);
+  }
+
+  FailureOr<TilingResult>
+  generateResultTileValue(Operation *op, OpBuilder &b, unsigned resultNumber,
+                          ArrayRef<OpFoldResult> offsets,
+                          ArrayRef<OpFoldResult> sizes) const {
+
+    auto sliceOp = cast<tcp::SliceOp>(op);
+    auto loc = op->getLoc();
+
+    auto fold = [&](OpFoldResult ofr) {
+      return getValueOrCreateConstantIndexOp(b, loc, ofr);
+    };
+    auto add = [&](OpFoldResult val1, OpFoldResult val2) {
+      return b.createOrFold<arith::AddIOp>(loc, fold(val1), fold(val2));
+    };
+    auto mul = [&](OpFoldResult val1, OpFoldResult val2) {
+      return b.createOrFold<arith::MulIOp>(loc, fold(val1), fold(val2));
+    };
+
+    // Offset on the input of slice is computed by:
+    //     start + offset_on_output * stride
+    //
+    // Tile and fuse algorithm does not work with stride != 1.
+    // In order to get around this, we extract a contiguous chunk with
+    // `tensor.extract_slice` and use the following `tcp.slice` op to
+    // extract the strided parts. So, the size on the
+    // `tensor.extract_slice` will be
+    //     size * stride
```

Comment:
Actually, now that I think of it, the contiguous size can be slightly smaller right? For instance, if the final slice `size` is just 1 element for a `tcp.slice` with a stride of 10, we'd still want to only extract 1 element given that we are already accounting for the correct starting offset. So the contiguous size can be:

```
newSize = (size - 1)*stride + 1
```

Reply:
I agree that will be slightly smaller. But would doing 2 extra operations (a sub and a add) be better than extracting the remaining elements as well? I'm not sure if that would have any benefits.

We can probably try this if needed later, if we see value in it. Wdyt?

Reply:
Sounds good. Maybe a little comment saying something to that effect?

Reply:
Added a comment